tests/jvm/src/test/resources/metac.expect

```diff
@@ -810,8 +810,7 @@ example.Example.main().(args) => param args: Array[String]
 example.Example.x(). => val method x: ClassTag[Int]
   ClassTag => scala.reflect.ClassTag#
   Int => scala.Int#
-local0 => selfparam self: Example
-  Example => example.Example.
+local0 => selfparam self: <?>
```

Comment:
Because previously we were saving synthetic self-type assigned to selfparams by scalac. In this PR, I found a way to separate the synthetic part from what's actually written in the source code. Now, selfparams without explicit type information have no signature.